clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -53,6 +53,31 @@ const char *SYCL::Linker::constructLLVMSpirvCommand(Compilation &C,
   return OutputFileName;
 }
 
+void SYCL::constructLLVMForeachCommand(Compilation &C, const JobAction &JA,
+                                       std::unique_ptr<Command> InputCommand,
+                                       ArgStringList &ForeachArgs,
+                                       const InputInfo &Output, const Tool *T) {
+  // Construct llvm-foreach command.
+  // The llvm-foreach command looks like this:
+  // llvm-foreach --in-file-list=a.list --in-replace='{}' -- echo '{}'
+  std::string OutputFileName(Output.getFilename());
+  ForeachArgs.push_back(
+      C.getArgs().MakeArgString("--out-file-list=" + OutputFileName));
```

Comment:
What if `InputCommand' has '"--out-file-list="' set as well?

Reply:
Do you mean if input command contains same argument?
All the arguments for `llvm-foreach` before `--` are treated as arguments for `llvm-foreach` itself, all the arguments after `--` are treated as input command parts, so it shouldn't be a problem...

Reply:
Oh, sorry, I rather meant "What if `ForeachArgs' already has '"--out-file-list="' in it?"

Reply:
Then `llvm-foreach` won't like it.

Reply:
This may be an additional reason to consider reworking the function so that it receives `llvm-foreach`-specific parameters in the form of the separate function arguments and not as an `ArgStringList`. Internally, `constructLLVMForeachCommand()` should know better.

Reply:
Yes, it seems you are right. I'll rework `constructLLVMForeachCommand()` function.

Reply:
Reworked.